### PR TITLE
Change tab width for .go files to 4

### PR DESCRIPTION
**Pull request by silverwind**
I thing 8-wide tabs are kind of archaic and inconsistent with template files which already use 4-wide.

Also did some misc cleanups in the .editorconfig files.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,4 +1,3 @@
-# http://editorconfig.org
```
Comment (6543) [inline .editorconfig:1]: why remove this?

**Comment by silverwind** [comment]
Looks like `gofmt` [hardcodes](https://github.com/golang/go/blob/2b70ffe9307c0992e28513ba25081d767b5937b2/src/cmd/gofmt/gofmt.go#L42) tab width to 8, so I'm not sure it's good to change it because if gofmt outputs spaces for some cases (I don't know) those would be misaligned.

File: .editorconfig (hunk 1 above)
Comment (silverwind) [reply .editorconfig:1]: Seemed like superflous advertising.

Comment (6543) [reply .editorconfig:1]: https://editorconfig.org/ <- we need our logo on it too ;)

@silverwind "... advertising ..." - OK I'm fine with it. you can find this by searching with keyword "editorconfig" too ...

**Comment by silverwind** [comment]
Based on https://github.com/golang/go/commit/25b51810014cefb8dba31321fcf40eb1a008fc3e, I think it's fine we change this. It says tab width has no effect on alignment of:

> Since "columns of alignment" are terminated whenever indentation changes from one line to the next, alignment with spaces will work independent of the actually chosen tab width

**Review by 6543 — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**